install/kubernetes/helm/istio/charts/grafana/templates/gateway.yaml

```diff
@@ -0,0 +1,44 @@
+{{- if .Values.ingressgateway.enabled -}}
+{{- $serviceName := include "grafana.fullname" . -}}
+{{- $servicePort := .Values.service.externalPort -}}
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: grafana-gateway
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "{{- .Values.ingressgateway.host }}"
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: grafana
+spec:
+  host: {{ template "grafana.name" . }}.{{ .Release.Namespace }}.svc.cluster.local
+  trafficPolicy:
+    tls:
+      mode: DISABLE
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: fortio
+spec:
+  hosts:
+  - "{{- .Values.ingressgateway.host }}"
+  gateways:
+  - grafana-gateway
+  http:
+  - route:
+    - destination:
+        port:
+          number: {{ $servicePort }}
+        host: {{ template "grafana.name" . }}.{{ .Release.Namespace }}.svc.cluster.local
```

Comment:
As I mentioned in a different PR - pilot, mixer, etc are attempting to minimize the number of options that are exposed to users. The name of the grafana service - grafana - is very unlikely to be something users will 
want to modify, and very likely to show up in non-helm configs or constants.

ie. replace template grafana.name with 'grafana'.

Reply:
done.